apps/ledger-live-mobile/src/screens/SyncOnboarding/index.tsx

```diff
@@ -310,33 +314,42 @@ export const SyncOnboarding = ({
       return;
     }
 
+    // case DeviceOnboardingStep.SafetyWarning not handled so the previous step (new seed, restore, recover) is kept
     switch (deviceOnboardingState?.currentOnboardingStep) {
+      case DeviceOnboardingStep.WelcomeScreen1:
+      case DeviceOnboardingStep.WelcomeScreen2:
+      case DeviceOnboardingStep.WelcomeScreen3:
+      case DeviceOnboardingStep.WelcomeScreen4:
+      case DeviceOnboardingStep.WelcomeScreenReminder:
+      case DeviceOnboardingStep.ChooseName:
+        setCompanionStepKey(CompanionStepKey.Paired);
+        break;
+      case DeviceOnboardingStep.Pin:
+        setCompanionStepKey(CompanionStepKey.Pin);
+        break;
       case DeviceOnboardingStep.SetupChoice:
-      case DeviceOnboardingStep.SafetyWarning:
         setCompanionStepKey(CompanionStepKey.Seed);
-        setSeedStatus("selection");
+        setSeedPathStatus("choice_new_or_restore");
         break;
       case DeviceOnboardingStep.NewDevice:
       case DeviceOnboardingStep.NewDeviceConfirming:
         setShouldRestoreApps(false);
         setCompanionStepKey(CompanionStepKey.Seed);
-        setSeedStatus("new");
+        setSeedPathStatus("new_seed");
+        break;
+      case DeviceOnboardingStep.SetupChoiceRestore:
+        setCompanionStepKey(CompanionStepKey.Seed);
+        setSeedPathStatus("choice_restore_direct_or_recover");
         break;
       case DeviceOnboardingStep.RestoreSeed:
         setShouldRestoreApps(true);
         setCompanionStepKey(CompanionStepKey.Seed);
-        setSeedStatus("restore");
-        break;
-      case DeviceOnboardingStep.WelcomeScreen1:
-      case DeviceOnboardingStep.WelcomeScreen2:
-      case DeviceOnboardingStep.WelcomeScreen3:
-      case DeviceOnboardingStep.WelcomeScreen4:
-      case DeviceOnboardingStep.WelcomeScreenReminder:
-      case DeviceOnboardingStep.ChooseName:
-        setCompanionStepKey(CompanionStepKey.Paired);
+        setSeedPathStatus("restore_seed");
         break;
-      case DeviceOnboardingStep.Pin:
-        setCompanionStepKey(CompanionStepKey.Pin);
+      case DeviceOnboardingStep.RecoverRestore:
+        setShouldRestoreApps(false);
```

Comment:
why false?

Reply:
because

Reply:
changing to true after our discussion ;)